services/impl/src/main/java/org/exoplatform/automatic/translation/impl/connectors/GoogleTranslateConnector.java

```diff
@@ -16,12 +16,96 @@
+
+  private static final String GOOGLE_TRANSLATE_SERVICE = "google-translate";
+
+  private static final String API_URL                  = "https://translation.googleapis.com/language/translate/v2";
+
+  private static final String KEY_PARAM                = "key";
+
+  private static final String DATA_PATTERN             = "{'q': '{message}', 'target': '{targetLocale}'}";
+
+  private static final int DEFAULT_POOL_CONNECTION = 100;
+
+  private HttpClient httpClient;
+
   public GoogleTranslateConnector(SettingService settingService) {
     super(settingService);
+
+
+    PoolingHttpClientConnectionManager connectionManager = new PoolingHttpClientConnectionManager();
+    connectionManager.setDefaultMaxPerRoute(DEFAULT_POOL_CONNECTION);
+    HttpClientBuilder httpClientBuilder = HttpClients.custom()
+                                                     .setConnectionManager(connectionManager)
+                                                     .setConnectionReuseStrategy(new DefaultConnectionReuseStrategy())
+                                                     .setMaxConnPerRoute(DEFAULT_POOL_CONNECTION);
+    this.httpClient = httpClientBuilder.build();
   }
+
+  @Override
+  public String translate(String message, Locale targetLocale) {
+    long startTime = System.currentTimeMillis();
+
+    String serviceUrl = API_URL + "?" + KEY_PARAM + "=" + getApiKey();
+    String data = DATA_PATTERN.replace("{message}", message).replace("{targetLocale}", targetLocale.getLanguage());
+    try {
+      HttpPost httpTypeRequest = new HttpPost(serviceUrl);
+      httpTypeRequest.setEntity(new StringEntity(data, ContentType.APPLICATION_JSON));
+      HttpResponse httpResponse=httpClient.execute(httpTypeRequest);
+      String response = null;
+      int statusCode = httpResponse.getStatusLine().getStatusCode();
+      if (statusCode == HttpURLConnection.HTTP_OK) {
+
+        // read the response
+        if (httpResponse.getEntity()!=null) {
+          try (InputStream is = httpResponse.getEntity().getContent()){
+            response = IOUtils.toString(is, StandardCharsets.UTF_8);
+          }
+        }
+
+        JSONObject jsonResponse = new JSONObject(response);
+        return jsonResponse.getJSONObject("data").getJSONArray("translations").getJSONObject(0).getString("translatedText");
+
+      } else {
+        LOG.error("remote_service={} operation={} parameters=\"message length:{},targetLocale:{}\" status=ko "
+                      + "duration_ms={} error_msg=\"Error sending translation request, status : {} \"",
+                  GOOGLE_TRANSLATE_SERVICE,
+                  "translate",
+                  message.length(),
+                  targetLocale.getLanguage(),
+                  System.currentTimeMillis() - startTime,
+                  statusCode);
+        return null;
+      }
+
+    } catch (Exception e) {
+      LOG.error("Error when trying to send translation request to google API");
```

Comment:
please avoid swallowing the original exception

Reply:
done